### ISSUE TITLE
Implement CIL-based just-in-time compiler

### ISSUE THREAD

**Issue by alexrp** [vezel-dev/niru#12]
Depends on #4 and #5.

The CIL-based JIT should be the first compilation-based execution module that is implemented. It will hopefully help reveal weaknesses in the IR for compilation scenarios. Additionally, it will provide low-effort JIT support for 32-bit host platforms, which is not a goal of the native JIT modules.

The implementation will likely just use `DynamicMethod`s.
